### PR TITLE
Add kube-api audit logging how-to

### PR DESCRIPTION
**Pull request by rosskirkpat**
This PR adds a guide on how to enable audit logging for the Kube API server.

[Preview](https://deploy-preview-906--docs-spectrocloud.netlify.app/audit-logs/kube-api-audit-logging)

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/906#" title="Author self-approved">rosskirkpat</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
Hi @rosskirkpat. Thanks for your PR.

I'm waiting for a [spectrocloud](https://github.com/orgs/spectrocloud/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/spectrocloud/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by spectro-prow** [comment]
@rosskirkpat: Cannot trigger testing until a trusted user reviews the PR and leaves an `/ok-to-test` message.

<details>

In response to [this](https://github.com/spectrocloud/librarium/pull/906#issuecomment-1270043318):

>/ok-to-test


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md
Comment (karl-cardenas-coding) [inline content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md:4]: Let's add a description that helps search engines identify the content and improve our SEO score.

File: content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md
Comment (karl-cardenas-coding) [inline content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md:16]: Let's keep titles shorter than five words, please.

File: content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md
Comment (karl-cardenas-coding) [inline content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md:18]: The introduction should be written in an active voice and explain to the reader what they are about to learn. Something along these lines.

Kubernetes is a powerful container orchestration system that contains many features, including the ability to review cluster usage through audit logs. In this tutorial, you will learn how to audit your Kubernetes cluster by using the audit logs provided by the kube-apiserver.

File: content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md
Comment (karl-cardenas-coding) [inline content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md:47]:
```suggestion
To start, you will use the [minimal audit policy file](#creating-the-audit-policy-file) to log all requests at the `metadata` level. 
```
Address the user through "you". Only use "we" when providing a recommendation our making a statement from us as a company.

File: content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md
Comment (karl-cardenas-coding) [inline content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md:57]:
```suggestion
On a control-plane node in the target cluster, issue the following command to create your audit policy file. 
```
It's best practice to avoid verbs based on physical traits (accessibility)

File: content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md
Comment (karl-cardenas-coding) [inline content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md:59]:
```suggestion
**NB.** This example configuration enables log truncation, compression, batching, and retention.
```
I'm not sure what NB is, but this can be added to the paragraph above instead of being a standalone line.

File: content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md
Comment (karl-cardenas-coding) [inline content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md:71]:
```suggestion
## Configure the kube-apiserver Manifest
```

File: content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md
Comment (karl-cardenas-coding) [inline content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md:115]: Instead of the text below, let's add this to a table to improve the readability. We can also avoid the callout in this line.

File: content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md
Comment (karl-cardenas-coding) [inline content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md:154]: Same as above, we can add this to the table.

File: content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md
Comment (karl-cardenas-coding) [inline content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md:156]: We should add a section called "Validate the Audit Logging" or something along those lines. We need to show the reader how we are validating everything was done correctly.

File: content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md
Comment (karl-cardenas-coding) [inline content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md:157]: We are missing a "Next Steps" section. The "Next Steps" section is the conclusion of the article. It starts by reviewing the key points the reader was taught and what they did. Lastly, it ends with suggestions for where to go next or other documentation.

**Review by karl-cardenas-coding — CHANGES_REQUESTED** [review]
Hi @rosskirkpat thank you so much for drafting this tutorial. There are a couple of things missing and changes we need to make to this tutorial. The main changes are related to text not following our [styling guide](https://spectrocloud.atlassian.net/wiki/spaces/DE/pages/1765933057/Spectro+Cloud+Internal+Style+Guide) and some readability and flow concerns. 

I'm more than happy to jump in a chat and help you work out these changes but here is an example of a [tutorial](https://spectrocloud.atlassian.net/wiki/spaces/DE/pages/1765933057/Spectro+Cloud+Internal+Style+Guide). 

I plan on releasing a Tutorial template for the organization soon to help people with drafting tutorials. Again, don't be sky and please reach out for help.

**Review by karl-cardenas-coding — APPROVED** [review]

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/906#pullrequestreview-1390655412" title="Approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/906#" title="Author self-approved">rosskirkpat</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/04.6-vm-management/15-vm-roles-permissions.md
Comment (rahulhazra97) [inline content/docs/04.6-vm-management/15-vm-roles-permissions.md:48]: To learn more, check out [Regex for Namespaces](/workspace/workload-features).

File: content/docs/10-audit-logs/05-kube-api-audit-logging.md
Comment (rahulhazra97) [inline content/docs/10-audit-logs/05-kube-api-audit-logging.md:33]: Enabling audit logging causes the API server to consume more memory, as it needs to store additional context for each request to facilitate auditing.

File: content/docs/10-audit-logs/05-kube-api-audit-logging.md
Comment (rahulhazra97) [inline content/docs/10-audit-logs/05-kube-api-audit-logging.md:51]: The Kubernetes API Audit policies define the rules for capturing events and specifying the level of detail to include.

File: content/docs/10-audit-logs/05-kube-api-audit-logging.md
Comment (rahulhazra97) [inline content/docs/10-audit-logs/05-kube-api-audit-logging.md:59]: Click on your cluster to access the details pages, and click on the **Nodes** tab. The tab contains information about each pool; select a node from the **Master Pool** to view its IP address.

File: content/docs/10-audit-logs/05-kube-api-audit-logging.md
Comment (rahulhazra97) [inline content/docs/10-audit-logs/05-kube-api-audit-logging.md:62]: SSH into one of your control-plane nodes using its IP address and the SSH key you specified during the cluster creation process.

File: content/docs/10-audit-logs/05-kube-api-audit-logging.md
Comment (rahulhazra97) [inline content/docs/10-audit-logs/05-kube-api-audit-logging.md:84]: Next, you will update the Kubernetes API server manifest file. The manifest file is in the **/etc/kubernetes/manifests** folder. 
Before you can modify the manifest file, please create a backup copy.

File: content/docs/10-audit-logs/05-kube-api-audit-logging.md
Comment (rahulhazra97) [reply content/docs/10-audit-logs/05-kube-api-audit-logging.md:93]: Now that you have a backup copy of the manifest file open the file **/etc/kubernetes/manifests/kube-apiserver.yaml** in a text editor such as Vi or Nano.

File: content/docs/10-audit-logs/05-kube-api-audit-logging.md
Comment (rahulhazra97) [inline content/docs/10-audit-logs/05-kube-api-audit-logging.md:125]: The top of the file contains the Kubernetes API parameters. Refer to the code snippet below to determine where to place these parameters.

File: content/docs/10-audit-logs/05-kube-api-audit-logging.md
Comment (rahulhazra97) [inline content/docs/10-audit-logs/05-kube-api-audit-logging.md:179]: Click on your cluster to access the details pages, and click on the **Nodes** tab. The tab contains information about each pool; select a node from the **Master Pool** to view its IP address.

File: content/docs/10-audit-logs/05-kube-api-audit-logging.md
Comment (rahulhazra97) [inline content/docs/10-audit-logs/05-kube-api-audit-logging.md:182]: SSH into one of your control-plane nodes using its IP address and the SSH key you specified during the cluster creation process.

File: content/docs/04.6-vm-management/15-vm-roles-permissions.md
Comment (karl-cardenas-coding) [inline content/docs/04.6-vm-management/15-vm-roles-permissions.md:48]:
```suggestion
Palette leverages Regex Pattern matching so you can select multiple namespaces to apply role bindings. To learn checkout [Regex for Namespaces](/workspace/workload-features).
```

File: content/docs/10-audit-logs/05-kube-api-audit-logging.md
Comment (karl-cardenas-coding) [inline content/docs/10-audit-logs/05-kube-api-audit-logging.md:33]:
```suggestion
Enabling audit logging causes the API server to consume more memory, as it needs to store additional context for each request to facilitate auditing. Memory consumption depends on the audit logging configuration.
```

File: content/docs/10-audit-logs/05-kube-api-audit-logging.md
Comment (karl-cardenas-coding) [inline content/docs/10-audit-logs/05-kube-api-audit-logging.md:51]:
```suggestion
The Kubernetes API Audit policies define the rules for capturing events and specifying the level of detail to include.
```

File: content/docs/10-audit-logs/05-kube-api-audit-logging.md
Comment (karl-cardenas-coding) [inline content/docs/10-audit-logs/05-kube-api-audit-logging.md:59]:
```suggestion
2. Identify one of your cluster contro-plane nodes. You can find a cluster node by navigating to the left **Main Menu** and selecting **Clusters**. Click on your cluster to access the details pages and click on the **Nodes** tab. The tab contains information about each pool, select a node from the **Master Pool** to view its IP address.
```

Comment (karl-cardenas-coding) [reply content/docs/10-audit-logs/05-kube-api-audit-logging.md:59]: I like your suggestion @rahulhazra97 but we need to tell the user how to get to the cluster details page. That's why I feel the current iteration is better to stick with.

File: content/docs/10-audit-logs/05-kube-api-audit-logging.md
Comment (karl-cardenas-coding) [inline content/docs/10-audit-logs/05-kube-api-audit-logging.md:62]:
```suggestion
3. SSH into one of your control-plane nodes using its IP address and the SSH key you specified during the cluster creation process.
```

File: content/docs/10-audit-logs/05-kube-api-audit-logging.md
Comment (karl-cardenas-coding) [inline content/docs/10-audit-logs/05-kube-api-audit-logging.md:85]:
```suggestion
Before you modify the manifest file, create a backup copy.
```

File: content/docs/10-audit-logs/05-kube-api-audit-logging.md
Comment (karl-cardenas-coding) [inline content/docs/10-audit-logs/05-kube-api-audit-logging.md:125]:
```suggestion
The top of the file contains the Kubernetes API parameters. Refer to the code snippet below to determine where to place these parameters.
```

File: content/docs/10-audit-logs/05-kube-api-audit-logging.md
Comment (karl-cardenas-coding) [inline content/docs/10-audit-logs/05-kube-api-audit-logging.md:182]:
```suggestion
3. SSH into one of your control-plane nodes using its IP address and the SSH key you specified during the cluster creation process. 
```

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/906#pullrequestreview-1390655412" title="Approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/906#" title="Author self-approved">rosskirkpat</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/release-3-4/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
